presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,203 @@
+            BaseJdbcConfig config,
+            OracleConfig oracleConfig,
+            ConnectionFactory connectionFactory)
+    {
+        super(config, "\"", connectionFactory);
+
+        this.fetchSize = oracleConfig.getFetchSize();
+
+        Set<String> tableTypes = new HashSet<>();
+        addAll(tableTypes, "TABLE", "VIEW");
+        if (oracleConfig.isIncludeSynonyms()) {
+            addAll(tableTypes, "SYNONYM");
+        }
+        this.tableTypes = tableTypes.toArray(new String[0]);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        String escape = metadata.getSearchStringEscape();
+        return metadata.getTables(
+                connection.getCatalog(),
+                escapeNamePattern(schemaName, escape).orElse(null),
+                escapeNamePattern(tableName, escape).orElse(null),
+                tableTypes.clone());
+    }
+
+    @Override
+    public PreparedStatement getPreparedStatement(Connection connection, String sql)
+            throws SQLException
+    {
+        PreparedStatement statement = connection.prepareStatement(sql);
+        statement.setFetchSize(fetchSize);
+        return statement;
+    }
+
+    @Override
+    protected String generateTemporaryTableName()
+    {
+        return "presto_tmp_" + System.nanoTime();
+    }
+
+    @Override
+    protected void renameTable(JdbcIdentity identity, String catalogName, String schemaName, String tableName, SchemaTableName newTable)
+    {
+        if (!schemaName.equalsIgnoreCase(newTable.getSchemaName())) {
+            throw new PrestoException(NOT_SUPPORTED, "Table rename across schemas is not supported in Oracle");
+        }
+
+        String sql = format(
+                "ALTER TABLE %s RENAME TO %s",
+                quoted(catalogName, schemaName, tableName),
+                quoted(newTable.getTableName()));
+
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+        switch (typeHandle.getJdbcType()) {
+            case Types.CLOB:
+                return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+            case Types.NUMERIC:
+                int precision = columnSize == 0 ? 38 : columnSize;
+                int scale = max(typeHandle.getDecimalDigits(), 0);
+                if (scale == 0) {
+                    if (precision <= TINYINT_NUMBER_PRECISION) {
+                        return Optional.of(tinyintColumnMapping());
```

Comment:
is this correct? TINYINT_NUMBER_PRECISION=3. i don't think all `decimal(3,0)` values can be represented in a tinyint

Reply:
you are right (all this values of incorrect), this should be ok for toWriteMapping but not for this method

but there are one problem, as we save all as number(p,s) and this will be mapped to bigInt or bigDecimal the test with integer will all fail
there are any way to check that a value from Types.NUMERIC is in the range of this values?